services/datamanager/builtin/builtin.go

```diff
@@ -610,28 +614,82 @@ func (svc *builtIn) Reconfigure(
 	if svc.syncSensor != syncSensor {
 		svc.syncSensor = syncSensor
 	}
-
 	syncConfigUpdated := svc.syncDisabled != svcConfig.ScheduledSyncDisabled || svc.syncIntervalMins != svcConfig.SyncIntervalMins ||
 		!reflect.DeepEqual(svc.tags, svcConfig.Tags) || svc.fileLastModifiedMillis != fileLastModifiedMillis ||
 		svc.maxSyncThreads != newMaxSyncThreadValue
 
 	if syncConfigUpdated {
+		svc.syncConfigUpdated = syncConfigUpdated
 		svc.syncDisabled = svcConfig.ScheduledSyncDisabled
 		svc.syncIntervalMins = svcConfig.SyncIntervalMins
 		svc.tags = svcConfig.Tags
 		svc.fileLastModifiedMillis = fileLastModifiedMillis
 		svc.maxSyncThreads = newMaxSyncThreadValue
+	}
+
+	// if datacapture is enabled, kick off a go routine to check if disk space is filling due to
+	// cached datacapture files
+	if !svc.captureDisabled {
+		fileDeletionCtx, cancelFunc := context.WithCancel(context.Background())
+		svc.fileDeletionRoutineCancelFn = cancelFunc
+		svc.fileDeletionBackgroundWorkers = &sync.WaitGroup{}
+		svc.fileDeletionBackgroundWorkers.Add(1)
+		go pollFilesystem(fileDeletionCtx, svc.fileDeletionBackgroundWorkers,
+			svc.captureDir, deleteEveryNthValue, svc.syncer, svc.logger)
+	}
+
+	g.Success()
+	return nil
+}
+
+func (svc *builtIn) startPropagateDataSyncConfig() {
+	svc.dataCaptureWorkers.Add(1)
+	goutils.ManagedGo(svc.propagateDataSyncConfigLoop, svc.dataCaptureWorkers.Done)
+}
 
+// propagateDataSyncConfigLoop runs until Close() is called on *builtIn
+// Immediately on first execution and every second afterwards it
+// checks if the datasync configuration has changes which
+// have not propagated to datasync.
+// If so it propagates the changes and marks the datasync configuration as propagated.
+// Otherwise it sleeps for another second.
+// Takes the builtIn lock every iteration.
+func (svc *builtIn) propagateDataSyncConfigLoop() {
+	if err := svc.propagateDataSyncConfig(); err != nil {
+		return
+	}
+	for goutils.SelectContextOrWait(svc.closedCtx, time.Second) {
+		if err := svc.propagateDataSyncConfig(); err != nil {
+			return
+		}
+	}
+}
+
+func (svc *builtIn) propagateDataSyncConfig() error {
+	svc.lock.Lock()
+	if svc.syncConfigUpdated {
```

Comment:
could you instead short-circuit this function?

```go
if !svg.syncConfigUpdated {
  return nil
}
// ...
```

Reply:
done